cmu_course_api/aggregate.py

```
@@ -37,25 +42,57 @@ def aggregate(schedules, fces):
     semester = SEMESTER_ABBREV[semester]
     year = schedules['semester'].split(' ')[-1][2:]
 
-    for course in schedules['schedules']:
+    count = cpu_count()
+    lock = threading.Lock()
+    isDone = False
+    isDoneLock = threading.Lock()
+    queue = Queue()
+
+    if count is None:
+        count = 4
+
+    print("running on " + str(count) + " threads")
+
+    def run():
+        while True:
+            try:
+                course = queue.get(timeout=4)
+                print('Getting description for ' + course['num'] + '...')
 
-        print('Getting description for ' + course['num'] + '...')
+                desc = get_course_desc(course['num'], semester, year)
+                desc['name'] = course['title']
 
-        desc = get_course_desc(course['num'], semester, year)
-        desc['name'] = course['title']
+                try:
+                    desc['units'] = float(course['units'])
+                except ValueError:
+                    desc['units'] = None
 
-        try:
-            desc['units'] = float(course['units'])
-        except ValueError:
-            desc['units'] = None
+                desc['department'] = course['department']
+                desc['lectures'] = course['lectures']
+                desc['sections'] = course['sections']
 
-        desc['department'] = course['department']
-        desc['lectures'] = course['lectures']
-        desc['sections'] = course['sections']
+                number = course['num'][:2] + '-' + course['num'][2:]
+                with lock:
+                    courses[number] = desc
+                queue.task_done()
+
+            except:
```

Can we only catch Empty exceptions here? (to make sure we're not unintentionally hiding other errors)